core/src/org/pentaho/di/core/row/value/ValueMetaBase.java

```diff
@@ -3220,7 +3220,8 @@ public String getDataXML( Object object ) throws IOException {
             // at all.
             //
             string = XMLHandler.addTagValue( "binary-string", (byte[]) object );
-            break;
+            xml.append( XMLHandler.openTag( XML_DATA_TAG ) ).append( string ).append( XMLHandler.closeTag( XML_DATA_TAG ) );
```

Comment:
@mbatchelor
XMLHandler.addTagValue( XML_DATA_TAG, string ) escapes XML specific tags < > 
and after that xml will be not correct parse, we will not get node <binary-string>